src/mlpack/core/data/load_csv.hpp

```diff
@@ -12,389 +42,276 @@
+  * the data for DatasetMapper, if MapPolicy::NeedsFirstPass is true.  The info
+  * object will be re-initialized with the correct dimensionality.
+  *
+  * @param rows Variable to be filled with the number of rows.
+  * @param cols Variable to be filled with the number of columns.
+  * @param info DatasetMapper object to use for first pass.
+  */
+  template<typename T, typename MapPolicy>
+  void InitializeMapper(size_t& rows, size_t& cols,
+                        DatasetMapper<MapPolicy>& info);
 
- private:
-  using iter_type = boost::iterator_range<std::string::iterator>;
+  /**
+  * Peek at the file to determine the number of rows and columns in the matrix,
+  * assuming a transposed matrix.  This will also take a first pass over the
+  * data for DatasetMapper, if MapPolicy::NeedsFirstPass is true.  The info
+  * object will be re-initialized with the correct dimensionality.
+  *
+  * @param rows Variable to be filled with the number of rows.
+  * @param cols Variable to be filled with the number of columns.
+  * @param info DatasetMapper object to use for first pass.
+  */
+  template<typename T, typename MapPolicy>
+  void InitializeTransposeMapper(size_t& rows, size_t& cols,
+                                 DatasetMapper<MapPolicy>& info);
 
   /**
-   * Check whether or not the file has successfully opened; throw an exception
-   * if not.
+   * Calculate number of columns in each row
+   * and assign the value to the col. This function
+   * will work for categorical data.
+   *
+   * @param lineStream a single row of data
+   * @param col number of columns in lineStream
+   * @param delim delimiter character
    */
-  void CheckOpen();
+  inline void CategoricalMatSize(std::stringstream& lineStream, size_t& col,
+                                 const char delim); 
+
+  // Functions common to both numeric & categorical parser
 
   /**
-   * Parse a non-transposed matrix.
+   * Get the size of the matrix. Based on isNumeric the fucntion can be used
+   * for both numeric_parse and categorical_parse.
    *
-   * @param inout Matrix to load into.
-   * @param infoSet DatasetMapper object to load with.
+   * @param f fstream stream to open the data file
+   * @param delim char delimiter charecter
    */
-  template<typename T, typename PolicyType>
-  void NonTransposeParse(arma::Mat<T>& inout,
-                         DatasetMapper<PolicyType>& infoSet)
+  template<bool isNumeric>
+  inline std::pair<size_t, size_t> GetMatrixSize(std::fstream& f,
+                                                 const char delim = ',')
   {
-    using namespace boost::spirit;
+    bool load_okay = f.good();
 
-    // Get the size of the matrix.
-    size_t rows, cols;
-    GetMatrixSize<T>(rows, cols, infoSet);
+    f.clear();
+    const std::fstream::pos_type pos1 = f.tellg();
 
-    // Set up output matrix.
-    inout.set_size(rows, cols);
-    size_t row = 0;
-    size_t col = 0;
+    size_t f_n_rows = 0;
+    size_t f_n_cols = 0;
```

Comment:
Yea, I think somehow the style checker is not working.

I think this directory is responsible for [style checking](https://github.com/mlpack/jenkins-conf/tree/master/linter), right?
I will have a look here, I would love to explore Jenkins.

Just let me know the right place.

I think @zoq tried something but it didn't work I guess.
This is happening since the start of the summer.

Reply:
Yep, exactly, it should run `lint.sh` and then post the output in Jenkins: https://github.com/mlpack/jenkins-conf/blob/master/linter/lint.sh

Maybe there is some bug or something causing it to not run right now; I'm not sure.